test/integration/contract/validateDPContractFactory.spec.js

```diff
@@ -451,6 +569,54 @@ describe('validateDPContractFactory', () => {
     expect(error.keyword).to.equal('additionalProperties');
   });
 
+  it('should return invalid result if there are duplicated indices', () => {
+    const indexDefinition = {
+      properties: {
+        $userId: 'asc',
+        firstName: 'desc',
+      },
+      unique: true,
+    };
+
+    rawDPContract.dpObjectsDefinition.indexedObject.indices.push(indexDefinition);
+
+    const result = validateDPContract(rawDPContract);
+
+    expectValidationError(result, DuplicateIndexError);
+
+    const [error] = result.getErrors();
+
+    expect(error.getIndexDefinition()).to.deep.equal(indexDefinition);
+  });
+
+  it('should return invalid result if indices don\'t have $userId prefix', () => {
+    const indexDefinition = rawDPContract.dpObjectsDefinition.indexedObject.indices[0];
+
+    delete indexDefinition.properties.$userId;
+
+    const result = validateDPContract(rawDPContract);
+
+    expectValidationError(result, UniqueIndexMustHaveUserIdPrefixError);
+
+    const [error] = result.getErrors();
+
+    expect(error.getIndexDefinition()).to.deep.equal(indexDefinition);
+  });
+
```

Comment:
I would test that $userId is not the first

Reply:
@shumkov Sorry, I've missed the point, you just want me to check it in test or add additional validation logic?

Reply:
@jawid-h We already have this logic. We need to test it.